src/main/java/com/datapipe/jenkins/vault/model/VaultSecretValue.java

```diff
@@ -40,24 +40,32 @@
     extends AbstractDescribableImpl<VaultSecretValue> {
 
     private String envVar;
+    private boolean isRequired;
     private final String vaultKey;
 
     @Deprecated
     public VaultSecretValue(String envVar, @NonNull String vaultKey) {
         this.envVar = fixEmptyAndTrim(envVar);
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = true; // Old behavior
     }
 
     @DataBoundConstructor
     public VaultSecretValue(@NonNull String vaultKey) {
         this.vaultKey = fixEmptyAndTrim(vaultKey);
+        this.isRequired = false;
```

Comment:
After testing this should preserve the old behavior
```suggestion
        this.isRequired = true;
```

Reply:
Now that I think about this will most likely always force it to true.

We should follow the pattern for defaults:
https://github.com/jenkinsci/notify-events-plugin/blob/3fbfb7b65d9dd72140692fc51d3da2911941a2a9/src/main/java/events/notify/NotifyEventsBuilder.java#L144-L149

Reply:
@jetersen done!